url_finder/src/url_tester.rs

```diff
@@ -149,6 +182,91 @@ pub async fn get_retrivability_with_head(urls: Vec<String>) -> (Option<String>,
     (sample_url, round_to_two_decimals(retri_percentage))
 }
 
+/// return retrivable percent of the urls
+pub async fn get_retrivability_with_get(urls: Vec<String>) -> (Option<String>, f64) {
+    let client = Client::builder()
+        .timeout(std::time::Duration::from_secs(RETRI_TIMEOUT_SEC))
+        .build()
+        .unwrap();
+    let success_counter = Arc::new(AtomicUsize::new(0));
+    let total_counter = Arc::new(AtomicUsize::new(0));
+
+    // stream of requests with concurency limit
+    let mut stream = stream::iter(urls)
+        .map(|url| {
+            let client = client.clone();
+            let total_clone = Arc::clone(&total_counter);
+            let success_clone = Arc::clone(&success_counter);
+            async move {
+                total_clone.fetch_add(1, Ordering::SeqCst);
+                match client.get(&url).send().await {
+                    Ok(resp) => {
+                        let content_type = resp
+                            .headers()
+                            .get("content-type")
+                            .and_then(|v| v.to_str().ok());
+                        let etag = resp.headers().get("etag");
+
+                        // check if the response has a content-type:
+                        // * application/octet-stream => curio
+                        // * application/piece        => boost
+                        // and an etag
+                        // in  header to indicating a file
+                        if resp.status().is_success()
+                            && matches!(
+                                content_type,
+                                Some("application/octet-stream") | Some("application/piece")
+                            )
+                            && etag.is_some()
+                        {
+                            tracing::info!("url WORKING: {:?}", url);
+                            success_clone.fetch_add(1, Ordering::SeqCst);
+                            Some(url)
+                        } else {
+                            debug!("Retrivability: URL::GET not working url: {:?}", url);
+                            None
+                        }
+                    }
+                    Err(err) => {
+                        debug!(
+                            "Retrivability: Get request for working url failed for {:?}: {:?}",
+                            url, err
+                        );
+                        None
+                    }
+                }
```

Comment:
This is big and complicated enough to extract and deduplicate with `filter_working_with_get`